src/module.c

```diff
@@ -1656,12 +1657,32 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
         return REDISMODULE_ERR;
     }
 
-    if (ReadConfig(ctx, argv, argc) == TSDB_ERROR) {
+    InitConfig();
+
+    const bool moduleConfigurationApiSupported = RTS_RedisSupportsModuleConfigApi();
+    if (ReadDeprecatedLoadTimeConfig(ctx, argv, argc, moduleConfigurationApiSupported) ==
+        TSDB_ERROR) {
         RedisModule_Log(
-            ctx, "warning", "Failed to parse RedisTimeSeries configurations. aborting...");
+            ctx,
+            "warning",
+            "Failed to parse RedisTimeSeries configurations using the deprecated way, aborting...");
```

Comment:
"Failed to parse deprecated legacy RedisTimeSeries configurations, aborting..."

"using the deprecated way" implies that what *we* are doing is deprecated.

Reply:
Indeed, this way of parsing the configurations is deprecated, and we won't be accepting the module configuration options passed as the module arguments any longer. So I meant exactly that and thought it would not be a problem. The parameters themselves aren't deprecated; the **way they are provided** is.

What do you think?

Reply:
I don't understand. The ability to pass load-time config options at all is deprecated? Even using the new, non-deprecated, names?